distribution.pm

```diff
@@ -282,10 +274,29 @@ sub script_output {
     }
 
     # and the markers including internal exit catcher
-    my $out = $output =~ /$marker(?<expected_output>.+)SCRIPT_FINISHED$marker-\d+-/s ? $+ : '';
-    # trim whitespaces
-    $out =~ s/^\s+|\s+$//g;
-    return $out;
+    my $out;
+    $out = $+{output} if ($output =~ /$marker(\r?\n|\025)(?<output>.*)SCRIPT_FINISHED$marker-$retval-/s);
+    return ($out, $retval);
+}
+
+=head2 script_output
+
+Wrap C<script_output_with_status()> and only return the trimmed output.
+Return empty string, if start- or end-markers were not found accordingly.
+=cut
+sub script_output {
+    my ($self, $script) = splice(@_, 0, 2);
+    my %args = testapi::compat_args(
```

Comment:
I would like to avoid the deprecated argument style in `distribution.pm`. Of course we keep it in `testapi.pm`.WDYT?

Reply:
well, yes. The `testapi.pm` should be considered our external API. Unfortunately this `script_output` might be used directly. Have you checked os-autoinst-distri-opensuse?

Reply:
There might be a problem in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/susedistribution.pm
It is based on `distribution` and calls sometime `$self->script_run()` https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/susedistribution.pm#L314

So I would keep it for now.

Reply:
There might be a problem in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/susedistribution.pm
It is based on `distribution` and calls sometime `$self->script_run()` https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/susedistribution.pm#L314

So I would keep it for now.